presto-main/src/main/java/io/prestosql/operator/DynamicFilterSourceOperator.java

```diff
@@ -0,0 +1,234 @@
+
+        public Channel(String filterId, Type type, int index)
+        {
+            this.filterId = filterId;
+            this.type = type;
+            this.index = index;
+        }
+    }
+
+    public static class DynamicFilterSourceOperatorFactory
+            implements OperatorFactory
+    {
+        private final int operatorId;
+        private final PlanNodeId planNodeId;
+        private final Consumer<TupleDomain<String>> dynamicPredicateConsumer;
+        private final List<Channel> channels;
+
+        public DynamicFilterSourceOperatorFactory(int operatorId, PlanNodeId planNodeId, Consumer<TupleDomain<String>> dynamicPredicateConsumer, List<Channel> channels)
+        {
+            this.operatorId = operatorId;
+            this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
+            this.dynamicPredicateConsumer = requireNonNull(dynamicPredicateConsumer, "dynamicPredicateConsumer is null");
+            this.channels = requireNonNull(channels, "channels is null");
+            verify(channels.stream().map(channel -> channel.filterId).collect(toSet()).size() == channels.size(),
+                    "duplicate dynamic filters are not allowed");
+            verify(channels.stream().map(channel -> channel.index).collect(toSet()).size() == channels.size(),
+                    "duplicate channel indices are not allowed");
+        }
+
+        @Override
+        public Operator createOperator(DriverContext driverContext)
+        {
+            return new DynamicFilterSourceOperator(
+                    driverContext.addOperatorContext(operatorId, planNodeId, DynamicFilterSourceOperator.class.getSimpleName()),
+                    dynamicPredicateConsumer,
+                    channels,
+                    planNodeId);
+        }
+
+        @Override
+        public void noMoreOperators()
+        {
+        }
+
+        @Override
+        public OperatorFactory duplicate()
+        {
+            throw new UnsupportedOperationException("duplicate() is not supported for DynamicFilterSourceOperatorFactory");
+        }
+    }
+
+    private final OperatorContext context;
+    private boolean finished;
+    private Page current;
+    private final Consumer<TupleDomain<String>> dynamicPredicateConsumer;
+    private final int maxFilterPositionsCount;
+    private final long maxFilterSizeInBytes;
+
+    private final List<Channel> channels;
+
+    // May be dropped if the predicate becomes too large.
+    private BlockBuilder[] blockBuilders;
+    private TypedSet[] valueSets;
+
+    private int filterPositionsCount;
+    private long filterSizeInBytes;
+
+    public DynamicFilterSourceOperator(OperatorContext context, Consumer<TupleDomain<String>> dynamicPredicateConsumer, List<Channel> channels, PlanNodeId planNodeId)
+    {
+        this.context = requireNonNull(context, "context is null");
+        this.maxFilterPositionsCount = getDynamicFilteringMaxPerDriverValueCount(context.getSession());
```

Comment:
calling this methods in constructor will affect benchmarking results as they parse strings. Pass these params to `DynamicFilterSourceOperatorFactory`.

Reply:
Good catch, done.